### DISCUSSION TITLE
tread break detection for embroidery mode

### DISCUSSION THREAD

**Discussion by Ams-Laser** [grblHAL/core#628] [Ideas]
can any one help me implement tread break detection for embroidery
tnks
Ed

**Comment by terjeio** [comment]
Both the physical detector and code to support it?

**Reply by Ams-Laser** [reply]
The phisical detector is already in place, just code is needee

**Reply by terjeio** [reply]
I have allowed myself a short break from coding. So be patient.

**Reply by Ams-Laser** [reply]
no problem, finally i got to test the embroidery plugin, there is an issue with handling jumps but works nice, im currently testing using M66 and is working good
thanks

**Reply by terjeio** [reply]
How should the controller handle thread breaks? Feed hold and stop spindle? Halt and alarm? ...?

**Reply by Ams-Laser** [reply]
the tread break circuitry is a pulse train generator, each stitch creates a pulse if you stop getting this pulse after X (programmable) stitches then asume you have a tread break and feedhold stop spindle

**Reply by Ams-Laser** [reply]
you also need a "JUMP SOLENOID"output line, (can be the flood or mist line) this must activate once the next stitch is detected as a jump, preferably before the previous stitch completes (once solenoid is engaged the next stitch wont happen as the needle is disengaged)

**Reply by terjeio** [reply]
When should the jump solenoid be deactivated? When the jump motion is completed?

**Reply by terjeio** [reply]
Here are some tentaive code changes, there are new settings for thread break input and jump solenoid output ports.
If the solenoid output port is set to -1 the mist output is used.
I am not able to test or debug these changes so you will have to do it.

[embroidery.zip](https://github.com/user-attachments/files/18825526/embroidery.zip)

**Reply by Ams-Laser** [reply]
> When should the jump solenoid be deactivated? When the jump motion is completed?
here is a sample of the gcode im using for a single jump stitch

(jmp)
 M3
 M66 P1 L1 Q30
 G00 X0.000 Y-5.200
 M5
 (jmp END)

it might be easyer to configure if we allow settings for gcode commands instead of hardcoding to certain outputs
something like:
JMP ON = M3
JMP OFF = M5

**Reply by Ams-Laser** [reply]
also note that the next movement after a needle change should be a jump stitch, also when a needle set command is encoutered a configurable output would be desired since either we need to pause for a manual change or we can use another axis motor to set the color changer position

**Reply by Ams-Laser** [reply]
> Here are some tentaive code changes, there are new settings for thread break input and jump solenoid output ports. If the solenoid output port is set to -1 the mist output is used. I am not able to test or debug these changes so you will have to do it.
> 
> [embroidery.zip](https://github.com/user-attachments/files/18825526/embroidery.zip)

i cant get IDE to compile, can you please compile for me and send ? sorry for the hassle
Ed
